src/cc/usdt/usdt.cc

```diff
@@ -273,6 +279,10 @@ std::string Context::resolve_bin_path(const std::string &bin_path) {
     ::free(which_so);
   }
 
+  if (!result.empty() && pid_) {
+    result = tfm::format("/proc/%d/root%s", *pid_, result);
```

Comment:
I think this is a bit debatable. Imagine this case. User wants to trace an USDT in `libc`, so user provided the path to `libc` (in root fs), as well as a PID. Now that PID has a different mount namespace that `libc` may live in a different path. This would actually cause that to fail.

@yonghong-song what do you think?

Reply:
I think this would be broken in the `setns` world as well? But not 100% sure I understand the scenario you're describing. Let me try to restate it.

This code assumes that absolute paths provided to `Context::resolve_bin_path` are relative to the root of the process being examined. If I want to trace USDTs in `/some/path/libc` in process `PID`, it's assumed that I'm talking about `/proc/PID/root/some/path/libc`, but I may actually be talking about a different `libc` that, as far as `PID` knows, lives at `/proc/PID/root/other/path/libc`, even though in the root ns it was at `/some/path/libc`.

If I'm understanding this right I would expect this to be broken in `setns` world. Not sure how to unbreak it going forward, but the fact that we're looking at `(device_id, inode)` starting with this diff is a good start.

Reply:
For usdt, if users provides both a pid and binary path, and the pid is the one outside the namespace, what should the binary path look look? In my opinion, user needs to provide
the binary path accessible outside the name space. Is this the right thing to do?

For example,
    inside namespace: pid = 4, path = /usr/lib/libc.so.6
    outside namespac: pid = 1234, path = /opt/name1/usr/lib/libc.so.6
If bpf tracing with usdt is done outside namespace, user parameters should be
    pid = 1234, path = /opt/name1/usr/lib/libc.so.6
                      or path = /proc/1234/root/usr/lib/libc.so.6
Do this sound right? 

We may not handle the following case properly if user says
  pid = 1234, library = c (libc)
In this case, bcc has to find out libc path inside the namespace and prefix it with
/proc/1234/root. Do we have a reliable way to get this without increasing complexity
too much? setns may not work if it is multi-threaded.
Otherwise, I am okay with forcing user to have a full path to avoid potential surprise.